src/System.Memory/tests/ReadOnlyBuffer/BufferSegment.cs

```diff
@@ -13,21 +13,50 @@ public BufferSegment(Memory<byte> memory)
             Memory = memory;
         }
 
-        /// <summary>
-        /// Combined length of all segments before this
-        /// </summary>
-        public long RunningIndex { get; private set; }
+        public long GetLength(IMemoryList<byte> memoryList)
+        {
+            BufferSegment current = this;
```

Comment:
Can we please rename this type and filename to include "Test" to avoid confusion with the real BufferSegment in S.IO.Pipelines?